fix: update and clarify Layout docs (#3651)
vaadin/docs: articles/flow/routing/layout.adoc

}
----

> The [annotationName]`@Layout` annotation path value needs to be unique for the project as else the resolver will not be able to determine which layout component should be used.

Reviewer: ```suggestion
The [annotationName]`@Layout` annotation path value needs to be unique for the project as else the resolver isn't able to determine which layout component should be used.
```

> Having multiple [annotationName]`@Layout` annotations with the same path value will throw an exception.

Reviewer: ```suggestion
Having multiple [annotationName]`@Layout` annotations with the same path value throws an exception.
```

> Opting out from automatically using a [interfacename]`RouterLayout` component annotated with [annotationname]`@Layout` the [annotationname]`@Route` and [annotationname]`@RouteAlias` contain the element `autoLayout` that can be set to false.

Reviewer: ```suggestion
Opting out from automatically using a [interfacename]`RouterLayout` component annotated with [annotationname]`@Layout` the [annotationname]`@Route` and [annotationname]`@RouteAlias` contain the annotation attribute `autoLayout` that can be set to false.
```

> This will ignore any automatic matching of layouts for the route.

Reviewer: ```suggestion
This ignores any automatic matching of layouts for the route.
```

> Giving a class for the `layout` method in [annotationname]`@Route` also disables automatic layouting and uses the given layout.

Reviewer: ```suggestion
Giving a class for the `layout` method in [annotationname]`@Route` also disables automatic layout and uses the given layout.
```
